### ISSUE TITLE
Performance Issue

### ISSUE THREAD

**Issue by tengzl33t** [lineageos-on-pdx201/releases#4]
There are some performance issues while running multiple apps like Spotify, Telegram, Firefox and Slack simultaneously. Interface is lagging and sound has some stuttering.
OS release: 15/06/2023
Magisk 26.1 and MindTheGapps-13.0.0-arm64-20230408_162909 installed.

**Comment by tengzl33t** [comment]
Android Auto restarts every 15-20 min, probably a problem with with ram.

**Comment by blackshibe** [comment]
Performance is noticeably better with the latest commits. I can multitask without any stutters.

**Comment by tengzl33t** [comment]
@blackshibe could you share the installable zip then, please?
@Mashopy or release is coming soon?

**Comment by blackshibe** [comment]
https://deadlinegame.com/lineage-20.0-20230721-UNOFFICIAL-pdx201.zip
This link will die when the web server updates.

Changes from original:
- i signed the build
- user type rather than userdebug
- signature spoofing enabled
- dual mono is disabled - i find the phasing effect it causes annoying
- mic gain is higher

**Comment by tengzl33t** [comment]
@blackshibe thanks, can i install it without wipes?

**Comment by blackshibe** [comment]
I don't know.

**Comment by tengzl33t** [comment]
@blackshibe can't install any additional packages, magisk or gapps, with your release, says that "can't mount /mnt/system" or "signature check failed".
@Mashopy could you release new build?

**Comment by blackshibe** [comment]
> @blackshibe can't install any additional packages, magisk or gapps, with your release, says that "can't mount /mnt/system" or "signature check failed". @Mashopy could you release new build?

i used an older twrp bringup to get around this without reading into it much
https://forum.xda-developers.com/t/unofficial-12-twrp-for-xperia-10-ii.4596879/

**Comment by tengzl33t** [comment]
> > @blackshibe can't install any additional packages, magisk or gapps, with your release, says that "can't mount /mnt/system" or "signature check failed". @Mashopy could you release new build?
> 
> i used an older twrp bringup to get around this without reading into it much https://forum.xda-developers.com/t/unofficial-12-twrp-for-xperia-10-ii.4596879/

Apologies, but could you confirm if this process is correct with this TWRP?:
1. Flash recovery provided by Mashopy
2. Flash copy partitions and reboot into LineageOS recovery
3. Flash your zip, reboot to fastboot and flash this TWRP
4. Reboot to TWRP, format data and flash Magisk and GApps 

As I have tried it a little bit too but unsuccessfully.

**Comment by Mashopy** [comment]
> @blackshibe can't install any additional packages, magisk or gapps, with your release, says that "can't mount /mnt/system" or "signature check failed". @Mashopy could you release new build?

Maybe in September, whenever I have enough free time

**Comment by blackshibe** [comment]
> > > @blackshibe can't install any additional packages, magisk or gapps, with your release, says that "can't mount /mnt/system" or "signature check failed". @Mashopy could you release new build?
> > 
> > 
> > i used an older twrp bringup to get around this without reading into it much https://forum.xda-developers.com/t/unofficial-12-twrp-for-xperia-10-ii.4596879/
> 
> Apologies, but could you confirm if this process is correct with this TWRP?:
> 
>     1. Flash recovery provided by Mashopy
> 
>     2. Flash copy partitions and reboot into LineageOS recovery
> 
>     3. Flash your zip, reboot to fastboot and flash this TWRP
> 
>     4. Reboot to TWRP, format data and flash Magisk and GApps
> 
> 
> As I have tried it a little bit too but unsuccessfully.

Yeah that's correct, I haven't ever gotten that error so I can't really help

Edit: 

I realized it might be because you're not using the recovery compiled with my devkeys? 
I uploaded the zip & recovery here:

https://github.com/blackshibe/android_device_sony_pdx201/releases/tag/20230721

**Comment by dedeweb** [comment]
@blackshibe Everything is fine except NFC, which doesn't work on my phone : https://github.com/lineageos-on-pdx201/releases/issues/6

**Comment by dedeweb** [comment]
NFC is working again, no idea what was the problem

**Comment by blackshibe** [comment]
Do you mind contacting me on discord or telegram?

**Comment by dedeweb** [comment]
after a few weeks of testing, I have to say I have performances issues too, even with @blackshibe  version. And I have the same problem as @tengzl33t : android auto is closing after a while. 
I also have frequent "application not responding" screens when I launch an app shortly after unlocking screen...

**Comment by Robotnik247** [comment]
I plan to build LOS and iodeOS for this device along with pdx-206 and pdx-203 next year, but I'm guessing zram is set to high? Because both XZ1 and XZ2 (and even other devices like the S8/S9) had this issue where if you set zram above 1024MB (say 2GB or even 4GB) things would start to lag and performance would degrade after only a few minutes. 

Currently pdx-206 doesn't have this problem even with zram set to 4GB (out of 8GB). I'm no hardware expert but my guess is that the RAM and SoC isn't fast enough to "decompress" and run from RAM which leads to said issues. SD 625 devices also had this issue which lead me to upgrade since 4GB RAM is kind of a bottleneck these days.

**Comment by Mashopy** [comment]
Just to inform I lost this device since months and I have no interest of maintaining this project any longer, even if I find it back.
